onnxscript/_framework_apis/torch_2_5.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Stable APIs for PyTorch 2.5."""
+
+from __future__ import annotations
+
+__all__ = [
+    "check_model",
+    "convert_version",
+    "get_torchlib_ops",
+    "optimize",
+    "save_model_with_external_data",
+    "OnnxFunctionMeta",
+]
+
+import dataclasses
+import os
+import pathlib
+from typing import Callable
+
+import onnx
+
+from onnxscript import ir
+from onnxscript.ir import _external_data
+
+# Internal flag. Will go away.
+_TORCH_ONNX_SAVE_EXTERNAL_DATA_WITH_IR = (
+    os.getenv("TORCH_ONNX_OFFLOAD_EXTERNAL_DATA_WITH_IR") == "1"
+)
+
+
+@dataclasses.dataclass(frozen=True)
+class OnnxFunctionMeta:
+    """A wrapper of onnx-script function with additional metadata.
+
+    qualified_name: The qualified name of the aten operator.
+    function: The onnx-script function.
+    domain: The domain of the function.
+    name: The name of the function.
+    is_complex: Whether the function is a complex function.
+    """
+
+    qualified_name: str
+    function: Callable
+    domain: str
+    name: str
+    is_complex: bool = False
+
+
+def optimize(model: ir.Model) -> ir.Model:
+    """Optimize the model."""
+
+    # TODO(justinchuby): Use the optimizer
+    return model
+
+
+def convert_version(model: ir.Model, target_version: int) -> ir.Model:
+    """Convert the model to the specified ONNX opset version."""
+    proto = ir.serde.serialize_model(model)
+    proto = onnx.version_converter.convert_version(proto, target_version)
+    return ir.serde.deserialize_model(proto)
+
+
+def check_model(model: ir.Model) -> None:
+    """Check the model."""
+
+    del model  # Unused yet
+
+
+def save_model_with_external_data(model: ir.Model, model_path: str | os.PathLike) -> None:
+    """Save the model with external data. The model is unchanged after saving."""
```

Comment:
For my education, when we do save model with external data, do we still expect the data in the model? This says the saved model still keeps the data, while the data is also saved externally.

Reply:
Yes. This way we can call save() multiple times.

Reply:
I thought we only save model with external data when the model is too large to be in one file?

Reply:
This is controlled by the ONNXProgram.save() api. It will decide when to call this function. When the model is saved, the ir.Model in the onnx program should not be altered, and hence we ensure this in this function.

Reply:
But if the model is too large, does it still save the very same size of model to one file? Would that hit protobuf limitation?

Reply:
When the model is too large. It will save to external tensors.